test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusProdModeTest.java

```diff
@@ -638,6 +625,7 @@ public void afterAll(ExtensionContext extensionContext) throws Exception {
         try {
             if (curatedApplication != null) {
                 curatedApplication.close();
+                curatedApplication = null;
```

Comment:
This change was probably enough to fix the issue, but when analyzing I found the timer one first and then this